### PR TITLE
mergify: add config yaml

### PR DESCRIPTION
**Pull request by SteVwonder**
@grondo: any idea how to add a list to the `approved-reviews-by` condition?  I attempted to use yaml arrays, but the mergify validation (`curl -F 'data=@.mergify.yml' https://gh.mergify.io/validate`) rejects it.

Both `- approved-reviews-by=[@flux-framework/core,@flux-framework/docs]` and

```yaml
- approved-reviews-by=
  - @flux-framework/core 
  - @flux-framework/docs
```

are rejected.

### CODE REVIEW

**Comment by grondo** [comment]
This seems like it may be a bug in mergify.io. I think the values in conditions are sent to the mergify engine as is (i.e. are not themselves yaml keys). The [documentation](https://github.com/Mergifyio/mergify-engine) does state that `approved-reviews-by` is an "array of string", but I think the teams `@` character is confusing the array parser.

I found [the simulator](https://dashboard.mergify.io/installation/661098/simulator?repositoryName=flux-core) useful, you can even run it against an open PR. I was able to make a list of users "work" with a simple comma-separated list, but if I added an @ character, it failed to parse.

The code seems to be here:

 https://github.com/Mergifyio/mergify-engine

maybe you can figure out what is going on.

**Comment by SteVwonder** [comment]
Thanks @grondo!  Poking around in the rules parser, it looks like they only allow a single team or username.  Not an array.  Relevant source line: https://github.com/Mergifyio/mergify-engine/blob/4d8f774d801355eb5aca6a8791d3b16367cdfab8/mergify_engine/rules/parser.py#L21

I opened an issue upstream to confirm: https://github.com/Mergifyio/mergify-engine/issues/656

Maybe we merge this PR as-is and get the multiple teams working later.

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+pull_request_rules:
+  - name: rebase and merge when passing all checks
+    conditions:
+      - base=master
+      - status-success=continuous-integration/travis-ci/pr
+      - status-success="validate commits"
```
Comment (grondo) [inline .mergify.yml]: Do we plan on enabling the pr-validator for this repo? If so maybe that can go in this PR as well. If not then the `"validate commits"` status can be removed from the mergify config.

Comment (SteVwonder) [reply .mergify.yml]: Great suggestion!  Added the action yml in this PR.

**Comment by SteVwonder** [comment]
I heard back on the upstream issue.  Looks like adding the second team requires a duplicate rule.  I'm waiting to hear back if the duplicate rule will somehow allow the action to be performed twice.

**Comment by SteVwonder** [comment]
It looks like duplicating the rule does cause the action to run twice.  So I think we stick with a single rule that uses `@flux-framework/docs`.  Just pushed that change.

**Comment by grondo** [comment]
Thanks @SteVwonder, merging.

**Pull request merged**